utility_analysis_new/combiners.py

```diff
@@ -349,3 +349,93 @@ def metrics_names(self) -> List[str]:
 
     def explain_computation(self):
         pass
+
+
+@dataclass
+class PrivacyIdCountUtilityAnalysisMetrics:
+    """Stores metrics for the privacy ID count utility analysis.
+
+    Args:
+        privacy_id_count: actual count of unique partitions contributed to by a privacy id.
+        std_noise: the noise standard deviation for DP count.
+        noise_kind: the type of noise used.
+    """
+    privacy_id_count: int
+    std_noise: float
+    noise_kind: pipeline_dp.NoiseKind
+
+
+@dataclass
+class UtilityAnalysisPrivacyIdCountAccumulator:
+    privacy_id_count: int
+    partition_selection_accumulator: Optional[PartitionSelectionAccumulator]
+
+    def __add__(self, other):
+        ps_accumulator = None
+        if self.partition_selection_accumulator is not None:
+            ps_accumulator = self.partition_selection_accumulator + other.partition_selection_accumulator
+
+        return UtilityAnalysisPrivacyIdCountAccumulator(
+            self.privacy_id_count + other.privacy_id_count, ps_accumulator)
+
+
+class UtilityAnalysisPrivacyIdCountCombiner(pipeline_dp.Combiner):
+    """A combiner for utility analysis privacy ID counts."""
+    AccumulatorType = UtilityAnalysisPrivacyIdCountAccumulator
+
+    def __init__(self, params: pipeline_dp.combiners.CombinerParams,
+                 is_public_partitions: bool):
+        self._params = params
+        self._is_public_partitions = is_public_partitions
+
+    def create_accumulator(self, data: Tuple[Sized, int]) -> AccumulatorType:
+        """Creates an accumulator for data.
+
+        Args:
+            data is a Tuple containing; 1) a list of the partitions a user contributed to, and 2) the user id.
+
+        Returns:
+            An accumulator with the privacy ID count of partitions and the contribution error.
+        """
+        if not data:
+            return UtilityAnalysisPrivacyIdCountAccumulator(0, None)
+        values, privacy_id = data
+        unique_partitions = set(list(values))
+        n_unique_partitions = len(unique_partitions)
```

Comment:
@dvadym This seems a little too simple, I was wondering about a way to store unique count _per_ partition, something like `count_per_partition = {partition : list(values).count(partition) for partition in unique_partitions}`, but I think I'd need a custom dictionary object because __ add __ isn't in dict object by default. WDYT?